source-dropbox/main.go

```diff
@@ -0,0 +1,220 @@
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	log "github.com/sirupsen/logrus"
+)
+
+type config struct {
+	Credentials credentials    `json:"credentials"`
+	Path        string         `json:"path"`
+	Parser      *parser.Config `json:"parser"`
+	MatchKeys   string         `json:"matchKeys,omitempty"`
+	Advanced    advancedConfig `json:"advanced"`
+}
+type advancedConfig struct {
+	AscendingKeys bool `json:"ascendingKeys,omitempty"`
+}
+
+type credentials struct {
+	ApiToken string `json:"apiToken"`
+}
+
+func (c config) Validate() error {
+	requiredProperties := [][]string{
+		{"AzureTenantID", c.Credentials.ApiToken},
+	}
+	for _, req := range requiredProperties {
+		if req[1] == "" {
+			return fmt.Errorf("missing '%s'", req[0])
+		}
+	}
+	return nil
+}
+
+func (c config) DiscoverRoot() string {
+	return c.Path
+}
+
+func (c config) RecommendedName() string {
+	return c.Path
+}
+
+func (c config) FilesAreMonotonic() bool {
+	return c.Advanced.AscendingKeys
+}
+
+func (c config) ParserConfig() *parser.Config {
+	return c.Parser
+}
+
+func (c config) PathRegex() string {
+	return c.MatchKeys
+}
+
+func newDropboxStore(ctx context.Context, cfg config) (*dropboxStore, error) {
+	httpClient, err := GetOAuth2HTTPClient(ctx, dropboxConfig)
+	if err != nil {
+		return &dropboxStore{}, err
+	}
+	config := dropbox.Config{
+		Client:   httpClient,
+		LogLevel: dropbox.LogInfo, // if needed, set the desired logging level. Default is off
+	}
+	client := files.New(config)
+
+	store := dropboxStore{
+		client: client,
+	}
+	if err := store.check(); err != nil {
+		return &dropboxStore{}, err
+	}
+	return &store, nil
+}
+
+type dropboxStore struct {
+	client files.Client
+	cfg    config
+}
+
+// check checks the connection to the Azure Blob Storage container.
+// It returns an error if the container is not found, the account is disabled, or if there is an authorization failure.
+// If the listing is successful, it returns nil.
+func (db *dropboxStore) check() error {
+	_, err := db.client.ListFolder(&files.ListFolderArg{})
+	if err != nil {
+		return fmt.Errorf("failed to list files: %w", err)
+	}
+	return nil
+}
+
+func (db *dropboxStore) List(ctx context.Context, query filesource.Query) (filesource.Listing, error) {
+	files, err := db.client.ListFolder(&files.ListFolderArg{Path: query.Prefix})
+	if err != nil {
+		return nil, err
+	}
+
+	return &dropboxListing{
+		ctx:    ctx,
+		client: db.client,
+		files:  *files,
+		index:  0,
+	}, nil
+}
+
+func (s *dropboxStore) Read(ctx context.Context, obj filesource.ObjectInfo) (io.ReadCloser, filesource.ObjectInfo, error) {
+	meta, reader, err := s.client.Download(&files.DownloadArg{Path: obj.Path})
+	if err != nil {
+		return nil, filesource.ObjectInfo{}, err
+	}
+
+	obj.ModTime = meta.ClientModified
+	obj.ContentType = meta.ExportInfo.ExportAs
+	obj.Size = int64(meta.Size) // Convert uint64 to int64
```

Comment:
The comment here isn't necessary - its very obvious what is happening from the code without the comment